h2o-algos/src/main/java/hex/tree/TreeHandler.java

```diff
@@ -62,10 +63,107 @@ else if (!(possibleModel instanceof SharedTreeModel) && !(possibleModel instance
+                languageRepresentation += getNewPaddedLine(padding);
+            }
+        }
+        return languageRepresentation;
+    }
+
+    private static String getNewPaddedLine(int padding) {
+        String line = "\n";
+        for(int i = 0; i < padding; i++) {
+            line += "\t";
+        }
+        return line;
+    }
+
+    private static String getElseLine(SharedTreeNode node) {
+        StringBuilder elseLine = new StringBuilder();
+        if (Float.compare(node.getSplitValue(),Float.NaN) != 0) {
+            elseLine.append("} else {");
+        } else {
+            SharedTreeNode leftChild = node.getLeftChild();
+            elseLine.append("} else if ( ").append(node.getColName()).append(" is in [ ");
+            String stringToParseInclusiveLevelsFrom = leftChild.getInclusiveLevels().toString();
+            int inclusiveLevelsLength = leftChild.getInclusiveLevels().toString().length();
+            if (inclusiveLevelsLength  > 2) {
+                // get rid of curly braces:
+                stringToParseInclusiveLevelsFrom = stringToParseInclusiveLevelsFrom.substring(1, inclusiveLevelsLength - 1);
+                String[] inclusiveLevels = stringToParseInclusiveLevelsFrom.split(",");
+                for (String index : inclusiveLevels) {
+                    elseLine.append(node.getDomainValues()[Integer.parseInt(index.trim())] + " ");
+                }
+            } else {
+                elseLine.append("Missing set of levels for underlying node");
+            }
+            elseLine.append("]) {");
+        }
+        return elseLine.toString();
+    }
+
+    private static String getConditionLine(SharedTreeNode node) {
+        StringBuilder conditionLine;
+        if (Float.compare(node.getSplitValue(),Float.NaN) != 0) {
+            conditionLine = new StringBuilder("If ( " + node.getColName() + " <= " + node.getSplitValue());
+            if ("RIGHT".equals(getNaDirection(node))) {
+                conditionLine.append(" or ").append(node.getColName()).append(" is NaN ) {");
+            } else {
+                conditionLine.append(" ) {");
+            }
+        } else {
+            conditionLine = new StringBuilder("If ( " + node.getColName() + " is in [ ");
+            // get inclusive levels:
+            SharedTreeNode rightChild = node.getRightChild();
+            String stringToParseInclusiveLevelsFrom = rightChild.getInclusiveLevels().toString();
+            int inclusiveLevelsLength = rightChild.getInclusiveLevels().toString().length();
+            if (inclusiveLevelsLength  > 2) {
+                // get rid of curly braces:
+                stringToParseInclusiveLevelsFrom = stringToParseInclusiveLevelsFrom.substring(1, inclusiveLevelsLength - 1);
+                String[] inclusiveLevels = stringToParseInclusiveLevelsFrom.split(",");
+                Arrays.stream(inclusiveLevels)
+                        .map(String::trim)
+                        .map(Integer::parseInt)
```

Comment:
```suggestion
                        .map(level -> Integer.parseInt(level.trim()))
```

One pass is better than two.

Reply:
I think zuzana's code is more reasable and the efficiency is the same

Reply:
After some hidden optimizations maybe, I'm not sure. The code shouts to map twice to me.

Reply:
would leave this as is

Reply:
The real-world experience would be the same, it's highly unlikely for anyone to notice even on larger trees.